frontend/packages/console-shared/src/components/pod/PodRing.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { VictoryLabel } from 'victory';
+import * as _ from 'lodash';
+import { Button, Split, SplitItem, Bullseye } from '@patternfly/react-core';
+import {
+  K8sResourceKind,
+  k8sPatch,
+  K8sKind,
+  SelfSubjectAccessReviewKind,
+} from '@console/internal/module/k8s';
+import { AngleUpIcon, AngleDownIcon } from '@patternfly/react-icons';
+import SvgDefsProvider from '../../../../dev-console/src/components/svg/SvgDefsProvider';
+import { calculateRadius, checkPodEditAccess } from '../../utils/pod-utils';
+import { Pod } from '../../types/topology-types';
+import PodStatus from './PodStatus';
+import './PodRing.scss';
+
+export interface PodRingProps {
+  pods: Pod[];
+  obj: K8sResourceKind;
+  resourceKind: K8sKind;
+  path: string;
+  impersonate?: string;
+}
+
+const PodRing: React.FC<PodRingProps> = ({ pods, obj, resourceKind, path, impersonate }) => {
+  const [editable, setEditable] = React.useState(false);
+  const ringContainerSize = 200;
+  const size = ringContainerSize - 80;
+  const { radius } = calculateRadius(size);
+
+  React.useEffect(() => {
+    checkPodEditAccess(obj, resourceKind, impersonate)
+      .then((resp: SelfSubjectAccessReviewKind) => setEditable(resp.status.allowed))
+      .catch((error) => {
+        throw error;
+      });
+  }, [pods, obj, resourceKind, impersonate]);
+
+  const handleScaling = (operation: number) => {
+    const patch = [{ op: 'replace', path, value: _.toInteger(obj.spec.replicas) + operation }];
```

Comment:
We may need to keep some internal state so that the buttons feel responsive and quickly clicking scale up / down buttons will not have to wait for the previous action to round trip.
eg. try to quickly click the scale up button. Observe it's not very responsive.

Maybe while there is a pending promise, use internal state. Once all promises have resolved, go back to using the incoming props.

cc @rohitkrai03